src/tests/blueprint/t_blueprint_mpi_mesh_transform.cpp

```diff
@@ -0,0 +1,381 @@
+    topo_0["type"] = "structured";
+    topo_1["type"] = "structured";
+
+    topo_0["elements/origin/i0"] = 0;
+    topo_0["elements/origin/j0"] = 0;
+    topo_1["elements/origin/i0"] = 8;
+    topo_1["elements/origin/j0"] = 0;
+    topo_0["elements/dims/i"] = 4;
+    topo_0["elements/dims/j"] = 4;
+    topo_1["elements/dims/i"] = 8;
+    topo_1["elements/dims/j"] = 8;
+
+    Node& field_0 = domain_0["fields/field"];
+    Node& field_1 = domain_1["fields/field"];
+    field_0["association"] = "element";
+    field_1["association"] = "element";
+    field_0["type"] = "scalar";
+    field_1["type"] = "scalar";
+    field_0["topology"] = "topo";
+    field_1["topology"] = "topo";
+
+    std::vector<double> vals_0(16, 0.5);
+    std::vector<double> vals_1(64, 1.5);
+    field_0["values"].set(vals_0);
+    field_1["values"].set(vals_1);
+
+    Node& adjset_0 = domain_0["adjsets/adjset"];
+    Node& adjset_1 = domain_1["adjsets/adjset"];
+    adjset_0["association"] =  "vertex";
+    adjset_0["topology"] =  "topo";
+    adjset_1["association"] =  "vertex";
+    adjset_1["topology"] =  "topo";
+
+    Node& group_0_1 = adjset_0["groups/group_000001"];
+    std::vector<int> nbrs_0(2);
+    nbrs_0[0] = 0;
+    nbrs_0[1] = 1;
+    group_0_1["neighbors"].set(nbrs_0);
+    group_0_1["rank"] = 0;
+
+    group_0_1["windows/window_000000/level_id"] = 0;
+    group_0_1["windows/window_000000/origin/i"] = 4;
+    group_0_1["windows/window_000000/origin/j"] = 0;
+    group_0_1["windows/window_000000/dims/i"] = 1;
+    group_0_1["windows/window_000000/dims/j"] = 5;
+    group_0_1["windows/window_000000/ratio/i"] = 2;
+    group_0_1["windows/window_000000/ratio/j"] = 2;
+    group_0_1["windows/window_000001/level_id"] = 1;
+    group_0_1["windows/window_000001/origin/i"] = 8;
+    group_0_1["windows/window_000001/origin/j"] = 0;
+    group_0_1["windows/window_000001/dims/i"] = 1;
+    group_0_1["windows/window_000001/dims/j"] = 9;
+    group_0_1["windows/window_000001/ratio/i"] = 2;
+    group_0_1["windows/window_000001/ratio/j"] = 2;
+
+    Node& group_1_0 = adjset_1["groups/group_000000"];
+    std::vector<int> nbrs_1(2);
+    nbrs_1[0] = 1;
+    nbrs_1[1] = 0;
+    group_1_0["neighbors"].set(nbrs_1);
+    group_1_0["rank"] = 0;
+
+    group_1_0["windows/window_000001/level_id"] = 1;
+    group_1_0["windows/window_000001/origin/i"] = 8;
+    group_1_0["windows/window_000001/origin/j"] = 0;
+    group_1_0["windows/window_000001/dims/i"] = 1;
+    group_1_0["windows/window_000001/dims/j"] = 9;
+    group_1_0["windows/window_000001/ratio/i"] = 2;
+    group_1_0["windows/window_000001/ratio/j"] = 2;
+    group_1_0["windows/window_000000/level_id"] = 0;
+    group_1_0["windows/window_000000/origin/i"] = 4;
+    group_1_0["windows/window_000000/origin/j"] = 0;
+    group_1_0["windows/window_000000/dims/i"] = 1;
+    group_1_0["windows/window_000000/dims/j"] = 5;
+    group_1_0["windows/window_000000/ratio/i"] = 2;
+    group_1_0["windows/window_000000/ratio/j"] = 2;
```

Comment:
As this example is a bit complicated, would it be possible to accompany it with an ASCII diagram in a code comment?

Reply:
Let's see if this works.  2 domains of differing resolutions, and the coarse elements at the interface will get extra vertices in the transformation.

```
_______________________
|    |    |__|__|__|__|
|____|____|__|__|__|__|
|    |    |__|__|__|__|
|____|____|__|__|__|__|
```